homeassistant/components/hue.py

```diff
@@ -257,3 +265,104 @@ def set_light(self, light_id, command):
     def set_group(self, light_id, command):
         """Change light settings for a group. See phue for detail."""
         return self.bridge.set_group(light_id, command)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class HueFlowHandler(config_entries.ConfigFlowHandler):
+    """Handle a Hue config flow."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize the Hue flow."""
+        self.host = None
+
+    @property
+    def _websession(self):
+        """Return a websession.
+
+        Cannot assign in init because hass variable is not set yet.
+        """
+        return aiohttp_client.async_get_clientsession(self.hass)
+
+    async def async_step_init(self, user_input=None):
+        """Handle a flow start."""
+        from aiohue.discovery import discover_nupnp
+
+        if user_input is not None:
+            self.host = user_input['host']
+            return await self.async_step_link()
+
+        bridges = await discover_nupnp(websession=self._websession)
+
+        if not bridges:
+            return self.async_abort(
+                reason='No Philips Hue bridges discovered.'
```

Comment:
Is the style guide that we end these messages with period? Ie not the same as log messages?

Reply:
Yes. It will be shown in the UI.

Reply:
It will be replaced with translation keys in a future release.